File: 파이썬/5.에라토스테네스의 체/4948.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/4948
 '''

l = [True] * 250000
l[1] = False
for i in range(2, 250000) :
    if l[i] == True :
        for j in range(i*2, 250000, i) : 
            l[j] = False

while True :
    n = int(input())
    if n == 0 :
        break
    cnt = 0
    for i in range(n+1, n*2+1) :
        if l[i] == True :
            cnt += 1
    print(cnt)

File: C++/017. 에라토스테네스의 체/4948.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/4948
*/

#include <iostream>
#include <algorithm>
#include <cmath>
#define MAX 250000

using namespace std;

int n;
bool check[MAX];

int main(){
    
    fill(check, check+MAX, true);
    check[1] = false;
    
    int s = sqrt(MAX);
    for(int i=2; i<=s; i++){
        if(check[i]){
            for(int j=i+i; j<=MAX; j+=i){
                check[j] = false;
            }
        }
    }
    
    while(1){
        cin >> n;
        if(n == 0){
            break;
        }
        int cnt = 0;
        for(int i=n+1; i<=n*2; i++){
            if(check[i]){
                cnt++;
            }
        }
        cout << cnt << "\n";
    }

    return 0;
}